src/app/_validation/PersonValidation.ts

```diff
@@ -1,12 +1,16 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
 import { PersonInput } from '../../interfaces';
-export const isValidFirstName = (firstName: string): boolean => firstName?.length > 0;
+export const isValidFirstName = (): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null =>
+        control.value?.length > 0
+            ? null : { strLen: control.value.length };
+}
 
-export const isValidPhone = (phone: string): boolean => /^\+?[0-9]{3}-?[0-9]{6,12}$/.test(phone);
+export const isValidPhone = (): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null => {
+        const isValid = /^\+?[0-9]{3}-?[0-9]{6,12}$/.test(control.value)
```

Comment:
oh, just missed it, 
let's take a look at validators that are coming from the library itself, 
you will find `pattern` where you can pass your regex, no need to make your own isValidPhone